NitroxClient/GameLogic/Constructor.cs

```diff
@@ -62,11 +64,50 @@ public void BeginCrafting(GameObject constructor, TechType techType, float durat
                 }
                 ConstructorBeginCrafting beginCrafting = new ConstructorBeginCrafting(constructorGuid, constructedObjectGuid, techType.Model(), duration, childIdentifiers, constructedObject.transform.position, constructedObject.transform.rotation, name, HSB, Colours);
                 packetSender.Send(beginCrafting);
+
+                // Mark vehicle as controlled by nitrox (used for sending add/remove batteries aka storage slots)
+                constructedObject.AddComponent<NitroxEntity>();
+                SendBatterySlots(constructedObject, childIdentifiers);
             }
             else
             {
                 Log.Error("Could not send packet because there wasn't a corresponding constructed object!");
             }
         }
+
+        private void SendBatterySlots(GameObject constructedObject, List<InteractiveChildObjectIdentifier> childIdentifiers)
```

Comment:
Why do we need to send the battery slots after the object begins crafting?  Shouldn't these already be taken care of by `InteractiveChildObjectIdentifiers`?

Reply:
SendBatterySlots will fill the StorageSlots with default batteries for the children, that are of EnergyMixin/BatterySource type. This needs to be done because i suppress the default batteries, that will be spawned for every vehicle.
I suppress them cause default batteries will be spawned for someone who connects. And these default batteries would cause many battery packets that will cause errors on other clients (cause guids are not synched) and also these default batteries shall not override the real batteries